cpp/src/arrow/flight/test-integration-server.cc

```diff
@@ -36,74 +37,27 @@ DEFINE_int32(port, 31337, "Server port to listen on");
 namespace arrow {
 namespace flight {
 
-class JsonReaderRecordBatchStream : public FlightDataStream {
- public:
-  explicit JsonReaderRecordBatchStream(
-      std::unique_ptr<ipc::internal::json::JsonReader>&& reader)
-      : index_(0), pool_(default_memory_pool()), reader_(std::move(reader)) {}
-
-  std::shared_ptr<Schema> schema() override { return reader_->schema(); }
-
-  Status Next(ipc::internal::IpcPayload* payload) override {
-    if (index_ >= reader_->num_record_batches()) {
-      // Signal that iteration is over
-      payload->metadata = nullptr;
-      return Status::OK();
-    }
-
-    std::shared_ptr<RecordBatch> batch;
-    RETURN_NOT_OK(reader_->ReadRecordBatch(index_, &batch));
-    index_++;
-
-    if (!batch) {
-      // Signal that iteration is over
-      payload->metadata = nullptr;
-      return Status::OK();
-    } else {
-      return ipc::internal::GetRecordBatchPayload(*batch, pool_, payload);
-    }
-  }
-
- private:
-  int index_;
-  MemoryPool* pool_;
-  std::unique_ptr<ipc::internal::json::JsonReader> reader_;
-};
-
 class FlightIntegrationTestServer : public FlightServerBase {
-  Status ReadJson(const std::string& json_path,
-                  std::unique_ptr<ipc::internal::json::JsonReader>* out) {
-    std::shared_ptr<io::ReadableFile> in_file;
-    std::cout << "Opening JSON file '" << json_path << "'" << std::endl;
-    RETURN_NOT_OK(io::ReadableFile::Open(json_path, &in_file));
-
-    int64_t file_size = 0;
-    RETURN_NOT_OK(in_file->GetSize(&file_size));
-
-    std::shared_ptr<Buffer> json_buffer;
-    RETURN_NOT_OK(in_file->Read(file_size, &json_buffer));
-
-    RETURN_NOT_OK(arrow::ipc::internal::json::JsonReader::Open(json_buffer, out));
-    return Status::OK();
-  }
-
   Status GetFlightInfo(const FlightDescriptor& request,
                        std::unique_ptr<FlightInfo>* info) override {
     if (request.type == FlightDescriptor::PATH) {
       if (request.path.size() == 0) {
         return Status::Invalid("Invalid path");
       }
 
-      std::unique_ptr<arrow::ipc::internal::json::JsonReader> reader;
-      RETURN_NOT_OK(ReadJson(request.path.back(), &reader));
+      auto data = uploaded_chunks.find(request.path[0]);
+      if (data == uploaded_chunks.end()) {
+        return Status::KeyError("Could not find flight.");
```

Comment:
`return Status::KeyError("Could not find flight ", request.path[0]);` would be a helpful output